README.md

```diff
@@ -610,29 +610,25 @@ You can also customize the behavior of `decode_request` and `encode_response` by
 overriding them in `LitAPI`. In this case:
 
 - `decode_request` takes a `litserve.specs.openai.ChatCompletionRequest` in input
-- `encode_response` returns a `litserve.specs.openai.ChatCompletionResponseChoice`
+- `encode_response` returns a `litserve.specs.openai.ChatMessage`
 
 See the OpenAI [Pydantic models](src/litserve/specs/openai.py) for reference.
 
 Here is an example of overriding `encode_response` in `LitAPI`:
 
 ```python
 import litserve as ls
-from litserve.specs.openai import ChatCompletionResponseChoice
+from litserve.specs.openai import ChatMessage
 
 class GPT2LitAPI(ls.LitAPI):
     def setup(self, device):
         self.model = None
 
     def predict(self, x):
-        return {"role": "assistant", "content": "This is a generated output"}
-
-    def encode_response(self, output: dict) -> ChatCompletionResponseChoice:
-        return ChatCompletionResponseChoice(
-            index=0,
-            message=ChatMessage(role="assistant", content="This is a custom encoded output"),
-            finish_reason="stop",
-        )
+        yield {"role": "assistant", "content": "This is a generated output"}
+
+    def encode_response(self, output: dict) -> ChatMessage:
+        yield ChatMessage(role="assistant", content="This is a custom encoded output")
```

Comment:
This is nicer, but by returning a `ChatMessage` we lose the ability to specify a `finish_reason`.
Right now it was hard-coded to `stop`, but we'll need to differentiate the cases eventually.